### ISSUE TITLE
Unsupported for grayscale image with 1 channel 1or 2D input?

### ISSUE THREAD

**Issue by Ford666** [VainF/pytorch-msssim#24]
It seems that whether the class SSIM/MS_SSIM or the function ssim/msssim are all unsupported for the input with 1 channel. When I fed the ssim function with input array of size (1,1,1984,1984), I encountered the following error:
`dim' is an invalid keyword argument for squeeze()`
What's more, only 4D numpy array is supported, but not for 2D image, which requires expanding dimension first before being processed. This is also not very convenient. Please consider adding the parameter for more versatile usage.

**Comment by VainF** [comment]
Hi @Ford666, please make sure that the input is a torch.Tensor.

**Comment by Ford666** [comment]
@VainF Yeah it works, thank you! So the input must be a torch.Tensor but without limit on its channel number.
